app/src/main/java/com/arygm/quickfix/ui/elements/DashBorder.kt

```diff
@@ -0,0 +1,222 @@
+                              cornerRadius = rrect.topLeftCornerRadius,
+                              style = stroke)
+                        } else if (insetPath != null) {
+                          drawPath(insetPath, brush, style = stroke)
+                        }
+                      }
+                      // Clip rect to ensure the stroke does not extend the bounds
+                      // of the composable.
+                      clipRect {
+                        // Draw a hairline stroke to cover up non-anti-aliased pixels
+                        // generated from the clip
+                        if (isSimpleRoundRect) {
+                          val rrect = (outline as Outline.Rounded).roundRect
+                          drawRoundRect(
+                              brush = brush,
+                              topLeft = Offset(rrect.left, rrect.top),
+                              size = Size(rrect.width, rrect.height),
+                              cornerRadius = rrect.topLeftCornerRadius,
+                              style =
+                                  Stroke(
+                                      Stroke.HairlineWidth,
+                                      pathEffect =
+                                          PathEffect.dashPathEffect(
+                                              floatArrayOf(on.toPx(), off.toPx()))))
+                        } else {
+                          drawPath(
+                              pathClip,
+                              brush = brush,
+                              style =
+                                  Stroke(
+                                      Stroke.HairlineWidth,
+                                      pathEffect =
+                                          PathEffect.dashPathEffect(
+                                              floatArrayOf(on.toPx(), off.toPx()))))
+                        }
+                      }
+                    } else {
+                      // Rectangular border fast path
+                      val strokeWidth = stroke.width
+                      val halfStrokeWidth = strokeWidth / 2
+                      drawRect(
+                          brush = brush,
+                          topLeft = Offset(halfStrokeWidth, halfStrokeWidth),
+                          size = Size(size.width - strokeWidth, size.height - strokeWidth),
+                          style = stroke)
+                    }
+                  }
+                }
+              })
+        },
+        inspectorInfo =
+            debugInspectorInfo {
+              name = "border"
+              properties["width"] = width
+              if (brush is SolidColor) {
+                properties["color"] = brush.value
+                value = brush.value
+              } else {
+                properties["brush"] = brush
+              }
+              properties["shape"] = shape
+            })
```

Comment:
I think this would be a bit cleaner, but of course not a must change:

```
inspectorInfo = debugInspectorInfo {
    name = "border"
    properties["width"] = width
    properties["shape"] = shape

    (brush as? SolidColor)?.let {
        properties["color"] = it.value
        value = `it.value`
    } ?: run {
        properties["brush"] = brush
    }
}
```
Everything else is very clean, great job.

Reply:
This is not my composable, I took it from this [Github Gist](https://gist.github.com/DavidIbrahim/236dadbccd99c4fd328e53587df35a21)